### PR TITLE
fix(webIDL): unambiguously link emum values

### LINKED ISSUE #1814: data-dfn-for ignored for duplicate IDL enum terms
## Important info

* URL to affected spec: [Presentation API](https://w3c.github.io/presentation-api/)
* ReSpec version: Click the ReSpec pill (e.g., 22.6.0).
* [x] I did a "hard refresh", but it's still busted. 

## Description of problem

The  `PresentationConnectionState` enum, defined in [section 6.5](https://w3c.github.io/presentation-api/#interface-presentationconnection) contains a `closed` term.
The `PresentationConnectionCloseReason` enum, defined in [section 6.5.4](https://w3c.github.io/presentation-api/#interface-presentationconnectioncloseevent) also contains a `closed` term.

Definitions use `data-dfn-for` to distinguish between the two, but ReSpec seems to ignore these attributes for `closed` (but not for the other terms). The result is that:
1. the prose definition of the `closed` term for `PresentationConnectionState`, whose id should be `dom-presentationconnectionstate-closed` ends up with the id `dom-presentationconnectionclosereason-closed`
3. the prose definition of the `closed` term for `PresentationConnectionCloseReason`, whose id should be `dom-presentationconnectionclosereason-closed` ends up with the id `dfn-closed` (and is not identified as an IDL term)

Minimal markup that reproduces the problem:

```html
<pre class="idl">
  enum State { "closed" };
  enum CloseReason { "closed" };
</pre>
<p>For <dfn>State</dfn>, <dfn data-dfn-for="State">closed</dfn> means foo.</p>
<p>For <dfn>CloseReason</dfn>, <dfn data-dfn-for="CloseReason">closed</dfn> means bar.</p>
```

I had not noticed that problem before so this may have been triggered by recent changes to IDL auto-linking code.

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #1814

### CODE REVIEW

**Comment by marcoscaceres** [comment]
There was a bug where enums (top level things) were being confused for enum values! That was causing the wrong `dfn` to be matched.

**Comment by marcoscaceres** [comment]
@sidvishnoi asked me to add some tests. @sidvishnoi can you take a look?

File: tests/spec/core/webidl.html [hunk 1]
```diff
@@ -599,6 +600,19 @@ <h2>Enumerations</h2>
         <p><dfn>not empty</dfn>...</p>
       </section>
     </section>
+    <section id="multpleEnums" data-dfn-for="Test2">
```
Comment (sidvishnoi) [inline tests/spec/core/webidl.html]: `multpleEnums` typo?

Comment (marcoscaceres) [reply tests/spec/core/webidl.html]: oopsy

Comment (marcoscaceres) [reply tests/spec/core/webidl.html]: fixed.

File: tests/spec/core/webidl-spec.js [hunk 2]
```diff
@@ -707,6 +707,17 @@ partial dictionary AnotherThing {
     expect(idl.querySelector(".idlSectionComment")).toBeNull();
   });
 
+  it("uniquely links to enum values", () => {
+    const target = doc.getElementById("multipleEnums");
+    const idlLinks = target.querySelectorAll("a[data-link-for]");
```
Comment (sidvishnoi) [inline tests/spec/core/webidl-spec.js]: Nit:
can probably just do `doc.querySelectorAll("#multipleEnums a[data-link-for]");`

**Pull request merged**